contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -712,42 +712,69 @@ PG_FUNCTION_INFO_V1(user_id);
 Datum
 user_id(PG_FUNCTION_ARGS)
 {
-	char	   *user_input;
-	char	   *user_name;
-	char	   *db_name;
-	HeapTuple	auth_tuple;
-	Form_pg_authid authform;
-	Oid			ret;
+    char    *user_input;
+    char    *user_name;
+    char    *db_name;
+    HeapTuple    auth_tuple;
+    Form_pg_authid   authform;
+    Oid     ret;
+    size_t  len;
 
-	user_input = PG_ARGISNULL(0) ? NULL : text_to_cstring(PG_GETARG_TEXT_PP(0));
-	db_name = get_cur_db_name();
+    user_input = PG_ARGISNULL(0) ? NULL : text_to_cstring(PG_GETARG_TEXT_PP(0));
+    db_name = get_cur_db_name();
 
-	if (!db_name)
-		PG_RETURN_NULL();
+    if (!db_name)
+        PG_RETURN_NULL();
 
-	if (!user_input)
-		PG_RETURN_OID(GetUserId());
+    if (!user_input)
```

Comment:
user_input could not be NULL since you defined the function as strict. 
Even if it can, expected result is NULL, not current user id .